linked_list/linked_list.py

```diff
@@ -86,10 +161,22 @@ def visit(self):
 
     # method to reverse the singly linked list
     # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)
     def reverse(self):

```

Comment:
👍 Nicely done, but the space complexity is O(1) because no matter how big the list is, you're only making 3 variables, none of which is a larger data structure.